raiden/ui/runners.py

```diff
@@ -140,14 +140,16 @@ def _start_services(self) -> None:
 
         self._startup_hook()
 
-        stop_event: AsyncResult[None] = AsyncResult()
+        stop_event: AsyncResult[Optional[signal.Signals]]  # pylint: disable=no-member
+        stop_event = AsyncResult()
 
-        def sig_set(sig: Any = None, _frame: Any = None) -> None:
-            stop_event.set(sig)
+        def sig_set(sig: int, _frame: Any = None) -> None:
+            stop_event.set(signal.Signals(sig))  # pylint: disable=no-member
 
-        gevent.signal(signal.SIGQUIT, sig_set)
-        gevent.signal(signal.SIGTERM, sig_set)
-        gevent.signal(signal.SIGINT, sig_set)
+        gevent.signal.signal(signal.SIGQUIT, sig_set)  # pylint: disable=no-member
+        gevent.signal.signal(signal.SIGTERM, sig_set)  # pylint: disable=no-member
+        gevent.signal.signal(signal.SIGINT, sig_set)  # pylint: disable=no-member
+        gevent.signal.signal(signal.SIGPIPE, sig_set)  # pylint: disable=no-member
```

Comment:
Can this happen easily? I've never heard of this one.

Reply:
Sigpipe is sent when one (or both) of the output file descriptors have been redirected to a pipe and that pipe is closed.

I’ve included it here to increase reliability when raiden is being run by some external process manager that does something to the output streams (e.g. SP, stress test, docker, etc.)

It’s also useful for local testing when piping logs into tools that close the input after some condition.
For example waiting for a specific log entry and automatically stop raiden afterwards:
```
raiden --log-json | grep --max-count 5 “Sync returned” | jq .
```